Dockerfile

```
@@ -1,6 +1,33 @@
-FROM bitnami/nginx:1.18.0
+FROM registry.access.redhat.com/ubi8/nodejs-14 as builder
+
+ENV NODE_OPTIONS=--max-old-space-size=8192
+
+ARG REACT_APP_GIT_SHA
+ENV REACT_APP_GIT_SHA=$REACT_APP_GIT_SHA
+ARG REACT_APP_VERSION
+ENV REACT_APP_VERSION=$REACT_APP_VERSION
+
+COPY --chown=1001:0 / /src/
+RUN chmod 775 /src/
+WORKDIR /src/
+
+RUN npx yarn install
+RUN npx yarn lint
+RUN npx yarn build
+
+FROM registry.access.redhat.com/ubi8/nginx-120
+
+# persist these on the final image for later inspection
+ARG REACT_APP_GIT_SHA
+ENV GIT_SHA=$REACT_APP_GIT_SHA
+ARG REACT_APP_VERSION
+ENV VERSION=$REACT_APP_VERSION
 
 COPY deploy/deploy_config.sh /deploy/
 COPY deploy/ui-deployment-template.yaml /deploy/
+COPY deploy/nginx.conf /deploy/
+COPY deploy/start.sh /deploy/
+
+COPY --from=builder /src/build/ "${NGINX_APP_ROOT}/src/"
 
-COPY build/ /app/
\ No newline at end of file
+CMD [ "/deploy/start.sh" ]
```

Following [this guie](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#entrypoint), wouldn't it be better to use the ENTRYPOINT directive in this case, WDYT?
```suggestion
ENTRYPOINT [ "/deploy/start.sh" ]
```

---

It depends. ENTRYPOINT is useful when you expect there to be arguments that a user could/should supply at runtime. The downside is that you cannot override the ENTRYPOINT at runtime. It is sometimes useful for development, testing, or troubleshooting to run different commands than the one specified in the Dockerfile, and this option prevents that.

Using just CMD gives you the option to specify a different command at runtime. Many images take that approach, including the UBI nginx base image as a relevant example.

It's not likely to make a big difference either way, but I would retain the flexibility of CMD unless there was something about the ENTRYPOINT we definitely want to take advantage of.

---

ack, 10x for the explanation